tests-extension/test/ingress_operator_util.go

```
@@ -0,0 +1,692 @@
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+func waitForICAvailable(ctx context.Context, dc dynamic.Interface, name string, timeout time.Duration) error {
+	return wait.PollUntilContextTimeout(ctx, pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		ic, err := dc.Resource(ingressControllerGVR).Namespace(operatorNamespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 173) but it returns nil

(nilerr)